lib/shelf-metadata.js

```diff
@@ -0,0 +1,261 @@
+                logger.debug("In request.post().then()");
+
+                return utils.handleResponse(response);
+            });
+        }
+
+
+        /**
+         * Deletes metadata property on the file.
+         *
+         * @param {string} property The metadata property to delete.
+         * @return {Promise.<Object>}
+         */
+        deleteProperty(property) {
+            var options;
+
+            logger.debug("In ShelfMetadata.deleteProperty()");
+            logger.debug(`Property param: ${property}`);
+
+            if (!property) {
+                throw new Error("Must provide a property to delete.");
+            }
+
+            options = this.getOptions(property);
+            logger.info(`Creating metadata for ${options.url}`);
+            logger.debug(`DELETE request options: ${JSON.stringify(options, null, 4)}`);
+
+            return requestPromise.delete(options)(options).then((response) => {
+                logger.debug("In request.delete().then()");
+
+                return utils.handleResponse(response);
+            });
+        }
+
+
+        /**
+         * @typedef {Object} options
+         * @property {string} url
+         * @property {Object} headers
+         * @proptery {Boolean} resolveWithFullResponse
+         * @property {Boolean} simple
+         * @property {string} rejectUnathorized
+         */
+
+        /**
+         * Creates and returns the options for requests.
+         *
+         * @param {string} property Optional file metadata property.
+         * @return {Object} options The options object used for requests.
+         */
+        getOptions(property) {
+            var options, url;
+
+            if (property) {
+                url = `${this.uri}/_meta/${property}`;
+            } else {
+                url = `${this.uri}/_meta`;
+            }
+
+            // NOTE: If you are using this library with Shelf with a self
+            // signed certificate, you will need to add the property
+            // "rejectUnauthorized: false" to this options object.
+            options = {
+                url,
+                headers: {
+                    Authorization: this.authToken
+                },
+                resolveWithFullResponse: true,
+                simple: false
+            };
+
+            return options;
+        }
+
+        /**
+         * @typedef {Object} options
+         * @property {string} url
+         * @property {Object} headers
+         * @property {string} rejectUnathorized
+         * @property {Object} body
+         * @property {Boolean} json
+         */
+
+        /**
+         * Sets and returns the options body property with the data added in.
+         *
+         * @param {Object} options The options object to set the data on.
+         * @param {Object} data The data object to set in the request options.
+         * @return {Object} options
+         */
+        setOptionsData(options, data) {
```

Comment:
Yeah. I can't remember why I wrote this. I obviously didn't think it through. Removing.